src/analysis/normed_space/star/multiplier.lean

```diff
@@ -0,0 +1,414 @@
+  star_add := λ x y, by {ext; simp only [star_left, star_right, add_left, add_right,
+    continuous_linear_map.add_apply, star_add]},
+  .. double_centralizer.has_star }
+
+instance : star_ring 𝓜(𝕜, A) :=
+{ star_mul := λ a b, by {ext; simp only [star_left, star_right, mul_left, mul_right, star_star,
+    continuous_linear_map.coe_mul, function.comp_app]},
+  .. double_centralizer.star_add_monoid }
+
+instance : star_module 𝕜 𝓜(𝕜, A) :=
+{ star_smul := λ k a, by {ext; exact star_smul _ _},
+  .. double_centralizer.star_add_monoid }
+
+end star
+
+/-!
+### Coercion from an algebra into its multiplier algebra
+-/
+
+noncomputable instance : has_coe_t A 𝓜(𝕜, A) :=
+{ coe := λ a,
+  { left := continuous_linear_map.mul 𝕜 A a,
+    right := (continuous_linear_map.mul 𝕜 A).flip a,
+    central := λ x y, mul_assoc _ _ _ } }
+
+@[simp, norm_cast]
+lemma coe_left (a : A) : (a : 𝓜(𝕜, A)).left = continuous_linear_map.mul 𝕜 A a := rfl
+@[simp, norm_cast]
+lemma coe_right (a : A) : (a : 𝓜(𝕜, A)).right = (continuous_linear_map.mul 𝕜 A).flip a := rfl
+
+section
+variables [star_ring 𝕜] [star_ring A] [star_module 𝕜 A] [normed_star_group A]
+/-- The coercion of an algebra into its multiplier algebra as a non-unital star algebra
+homomorphism. -/
+def non_unital_star_algebra_hom_coe : A →⋆ₙₐ[𝕜] 𝓜(𝕜, A) :=
+{ to_fun := λ a, a,
+  map_smul' := λ k a, by {ext1; simp only [coe_left, coe_right, continuous_linear_map.map_smul,
+    smul_left, smul_right]},
+  map_zero' := by {ext1; simp only [coe_left, coe_right, map_zero, zero_left, zero_right]},
+  map_add' := λ a b, by {ext1; simp only [coe_left, coe_right, map_add, add_left, add_right]},
+  map_mul' := λ a b, by {ext; simp only [coe_left, coe_right, mul_apply',
+    flip_apply, mul_left, mul_right, continuous_linear_map.coe_mul,
+    function.comp_app, mul_assoc]},
+  map_star' := λ a, by {ext; simp only [coe_left, mul_apply', star_left, star_right, coe_right,
+    flip_apply, star_mul, star_star]} }
+end
+
+/-!
+### Norm structures
+-/
+
+noncomputable instance : normed_ring 𝓜(𝕜, A) :=
+{ norm_mul := λ a b,
+    begin
+      refine max_le ((norm_mul_le _ _).trans _) ((norm_mul_le _ _).trans _),
+      exact mul_le_mul (le_max_left _ _) (le_max_left _ _) (norm_nonneg _)
+        ((norm_nonneg _).trans $ le_max_left _ _),
+      exact mul_comm (∥a.right∥) (∥b.right∥) ▸ mul_le_mul (le_max_right _ _) (le_max_right _ _)
+        (norm_nonneg _) ((norm_nonneg _).trans $ le_max_right _ _),
+    end,
+  .. double_centralizer.ring,
+  .. double_centralizer.normed_add_comm_group }
+
+noncomputable instance : normed_algebra 𝕜 𝓜(𝕜, A) :=
+{ ..double_centralizer.algebra, ..double_centralizer.normed_space }
+
+variables [star_ring A] [cstar_ring A]
+
+/-- For `a : 𝓜(𝕜, A)`, the norms of `a.left` and `a.right` coincide, and hence these
+also coincide with `∥a∥` which is `max (∥a.left∥) (∥a.right∥)`. -/
+lemma norm_left_eq_right (a : 𝓜(𝕜, A)) : ∥a.left∥ = ∥a.right∥ :=
```

Comment:
When reading I wondered whether you could half the proof by arguing by symmetry.

It's true that `𝓜(𝕜, A) ≃ 𝓜(𝕜, Aᵐᵒᵖ)`, right? With my suggestion of extending ` (A →L[𝕜] A) × (A →L[𝕜] A)`, this will be induced by `prod.swap`.

Reply:
I'm not sure it's that easy, unless I'm not understanding something. While the two directions are very similar, I don't see an obvious way to argue by symmetry, even via `Aᵐᵒᵖ`. At the very least, I think you would only be able to avoid `h2`, which is only 8 lines anyway.